addon/plugins/variable-plugin/utils/recreate-variable-uris.ts

```diff
@@ -0,0 +1,43 @@
+import { EXT } from '@lblod/ember-rdfa-editor-lblod-plugins/utils/constants';
+import { sayDataFactory } from '@lblod/ember-rdfa-editor/core/say-data-factory';
+import { OutgoingTriple } from '@lblod/ember-rdfa-editor/core/rdfa-processor';
+import { v4 as uuidv4 } from 'uuid';
+import { Attrs } from '@lblod/ember-rdfa-editor';
+
+export function recreateVariableUris(nodeAttrs: Attrs) {
+  const attrs = { ...nodeAttrs };
+  if (
+    attrs.subject &&
+    (attrs.subject as string).includes(
+      'http://data.lblod.info/mappings/--ref-uuid4-',
+    )
+  ) {
+    attrs.subject = `http://data.lblod.info/mappings/--ref-uuid4-${uuidv4()}`;
+  } else if (
+    attrs.subject &&
+    (attrs.subject as string).includes('http://data.lblod.info/mappings/')
+  ) {
+    attrs.subject = `http://data.lblod.info/mappings/${uuidv4()}`;
+  }
+  attrs.properties = (attrs.properties as OutgoingTriple[]).map((prop) => {
+    if (prop.predicate === EXT('instance').full) {
+      let recreatedUri;
+      if (
+        prop.object.value.includes(
+          'http://data.lblod.info/variables/--ref-uuid4-',
+        )
+      ) {
+        recreatedUri = `http://data.lblod.info/variables/--ref-uuid4-${uuidv4()}`;
```

Comment:
would it be better to use the library here? to make sure the behavior is consistent

Reply:
would also avoid the need for the includes check, so if we ever change the base it would keep working

Reply:
What do you mean? the template-uuid-instantiator? Is not what it does, here we recreate a uuid maintaining it being a template or a normal uri. The uuid instantiator transform from template to normal uuid

Reply:
oh yeah sorry, brainfart